objcgen/objcprocessor.cs

```diff
@@ -143,6 +144,13 @@ protected IEnumerable<MethodInfo> GetMethods (Type t)
 					continue;
 				}
 
+				if (implement_system_iequatable_t) {
+					if (mi.Match ("System.Boolean", "Equals", new string [] { null }) && !mi.Match ("System.Boolean", "Equals", "System.Object")) {
```

Comment:
you can avoid the 2nd `Match` if you move this into an `else` for the `Equals` check (next block)